docs/metrics.md

```diff
@@ -0,0 +1,29 @@
+# Metrics
+
+OCM Agent will expose the follow metrics via web server metric port `8383`.
+
+## Related resources
+
+`Service` and `ServiceMonitor`: Those are managed by the [OCM Agent operator](https://github.com/openshift/ocm-agent-operator/)
+
+`PrometheusRules`: Defines the alerting rules based on the metrics in
+[Managed Cluster Config](https://github.com/openshift/managed-cluster-config/)
+
+`Role` and `RoleBinding`: Required role and rolebinding to scrape the metrics from
+OCM Agent are managed by [OCM Agent operator](https://github.com/openshift/ocm-agent-operator/)
+
+## List of metrics
+
+|name|type|description|
+|----|----|----|
+|ocm_agent_requests_total|Counter|A total number of valid requests received by OCM Agent on web service port|
+|ocm_agent_requests_by_service|Counter|A total number of valid requests received by OCM Agent by different services,<br/>currently `alertmanager-receiver` is supported|
+|ocm_agent_failed_requests_total|Counter|A total number of failed requests received by OCM Agent on web service port|
+|ocm_agent_request_failure|Gauge|An indicator that when the OCM Agent cannot proceed the request successfully|
+|ocm_agent_response_failure|Gauge|An indicator that when the OCM Agent cannot get correct response from OCM services|
+|ocm_agent_service_log_sent|Counter|A total number of Service Log sent succeeded based the template name and firing state|
```

Comment:
I know @T0MASD was rather keen on these descriptions matching the descriptions that we use in the actual code. 
So with that in mind, we should either update our Help text in `metrics.go` to match these, or change these to match those (or a mix of both).

The proposal below would keep them in sync with the current Help descriptions.

```suggestion
|ocm_agent_requests_total|Counter|A count of total requests to ocm agent service|
|ocm_agent_requests_by_service|Counter|A count of total requests to ocm agent based on sub service|
|ocm_agent_failed_requests_total|Counter|A count of total failed requests received by the OCM Agent service|
|ocm_agent_request_failure|Gauge|Indicates that OCM Agent could not successfully process a request|
|ocm_agent_response_failure|Gauge|Indicates that the call to the OCM service endpoint failed|
|ocm_agent_service_log_sent|Counter|A count of total service log sent based on managedNotification template|
```